crates/bevy_render/src/renderer/mod.rs

```diff
@@ -97,13 +101,26 @@ pub struct RenderInstance(pub Instance);
 #[derive(Resource, Clone, Deref, DerefMut)]
 pub struct RenderAdapterInfo(pub AdapterInfo);
 
+/// The `AdapterInfo` of the adapter in use by the renderer.
+#[derive(Resource, Clone, Deref, DerefMut)]
+pub struct RenderTextureFormat(pub wgpu::TextureFormat);
+
+#[derive(Resource, Clone, Deref, DerefMut)]
+pub struct AvailableTextureFormats(pub Arc<Vec<wgpu::TextureFormat>>);
+
 /// Initializes the renderer by retrieving and preparing the GPU instance, device and queue
 /// for the specified backend.
 pub async fn initialize_renderer(
     instance: &Instance,
     options: &WgpuSettings,
     request_adapter_options: &RequestAdapterOptions<'_>,
-) -> (RenderDevice, RenderQueue, RenderAdapterInfo) {
+) -> (
```

Comment:
In my experience returning complex tuple types like this is a footgun and maintenance hazard. Can you refactor this to use a named struct?

Reply:
How about this:
```rust
// at bevy/crates/bevy_render/src/renderer/mod.rs
struct Renderer{
    render_device: RenderDevice,
    render_queue: RenderQueue,
    render_adapter_info: RenderAdapterInfo,
    render_adapter: RenderAdapter,
    available_texture_formats: AvailableTextureFormats,
}
```
Then make `initialize_renderer` return this struct and change all the related code?

Reply:
Yep, that's exactly the sort of thing I had in mind!

Reply:
Hi!  Maybe wrapping the return types into a struct over engineering? This function returns a tuple whose every element will be inserted as resource in the world. We only care about every element of the tuple. And before this change,  this function returns a three element tuple. I think theere is no need to change this. How do you think?

Reply:
Yeah I'm fine to leave it.